### ISSUE TITLE
recent update: issue with torchao 0.17.0  and 2.9.1+cu128

### ISSUE THREAD

**Issue by g3n3rativ3** [ostris/ai-toolkit#987]
edit: it is updated, with a rollback to a compatible torchao

The new update requires torchao 0.17.0, but there's an issue: 2.9.1+cu128 is still the required torch version for the app. But torchao 0.17.0 ask for torch 2.11.0
requirements_base.txt pins torchao==0.17.0, but the README installation instructions still specify torch==2.9.1+cu128. Per torchao's official compatibility table, torchao 0.17.0 requires torch >= 2.11.0 for its cpp extensions. This mismatch is likely the root cause.

First lines of the log when launching a training job:

Skipping import of cpp extensions due to incompatible torch version. Please upgrade to torch >= 2.11.0 (found 2.9.1+cu128).
Unable to import `torchao` Tensor objects. This may affect loading checkpoints serialized with `torchao`

Then there are hundreds of lines like:

Failed to quantize attn1: Float8Tensor dispatch: attempting to run unimplemented operator/function: func=<OpOverload(op='aten.abs', overload='default')>, types=(<class 'torchao.quantization.Float8Tensor'>,), arg_types=(<class 'torchao.quantization.Float8Tensor'>,), kwarg_types={}

Training still works despite this, but I'm not sure what the real impact on quantization is (VRAM usage seems unchanged on my end). Still, it would be good to resolve this properly.

**Comment by hwprinz** [comment]
Yes, exactly same here:

```
Skipping import of cpp extensions due to incompatible torch version. Please upgrade to torch >= 2.11.0 (found 2.9.1+cu128).
Unable to import `torchao` Tensor objects. This may affect loading checkpoints serialized with `torchao`
```

Happened with the uncontroled update of Toolkit to 0.12
this night

i mean think about it....
raising torchao to 0.17.0 but not taking torch itself into account 🤦 

torch 2.11 came into toolkit with the install script 0.5.3, means requiring a full new install of toolkit.
i installed toolkit with script 0.4.9. iirc it was even script 0.5.1 which still installed torch2.9.1......
so, very badly done this one...

Cheers

**Comment by FurkanGozukara** [comment]
the torchao version of ai toolkit is so old

i compiled latest torch 0.18 for cuda 13 and torch 2.13 and it broken ai toolkit

**Comment by hwprinz** [comment]
Hi Furkan
yes, correct, but 0.17.0 is just not running on a given installation
with downgrading the existing toolkit (torch2.9.1+cu128) to torchao 0.10.0 
brings it back to life

staying at 0.17.0  will break ANY training on that installation

of course, i have a 2nd install, installed with 0.5.3 install script, which loaded torch2.11.0+cu128 and had torchao 0.17.0 in it anyway.

but the main point to clearly make: as a dev, releasing smth updated, not should, but **HAS** to cover any existing installation
not only new ones. and just without telling anybody whats new in that release update?
(because we thankfully have such a great documentation and release notes of this Software 💪 )

 /Hans

**Comment by hwprinz** [comment]
> i compiled latest torch 0.18 for cuda 13 and torch 2.13 and it broken ai toolkit

Yes, i have another 3rd installation, which i had updated manually to torch2.12+cu130 + torchao 0.17.0
and some more 'needed' dependency packages.
that works still, whereby i have not yet updated to latest toolkit version, still on 0.11.2 (from 5 days ago)

general Statement:
nonetheless, its not clean anymore in any actual releases to train anything as before
as known in other issues, 40xx series cards are broken with it (as like on my 4060TI 16GB) (no power draw but 100% GPU)

and the actual installation of the update this morning, also broke a former running training job on my 5090, (also no power draw but 100% GPU)
i finally got it back running, by now quantizing the trainnig....
in older releases, a training was running some slower..... took longer, but, it was always running. now, its a mess

/Hans

**Comment by hwprinz** [comment]
@g3n3rativ3 what GPU are you running on, if i may ask?

**Comment by cushycrux** [comment]
python.exe -m pip uninstall torch torchvision, then
python.exe -m pip install -U torch torchvision --index-url https://download.pytorch.org/whl/cu130 and for sure
python.exe -m pip install -U torchaudio
removes the "torchao 0.17.0 requires torch >= 2.11.0" error on my win and 5070ti but introduces this errors:

> W0731 18:28:02.285000 3344 torch\utils\_pytree.py:630] <enum 'KernelPreference'> is an Enum subclass and is now natively supported by torch.compile as an opaque value type. Calling register_constant() on Enum subclasses is deprecated and will be an error in a future release.
> W0731 18:28:02.329000 3344 torch\utils\_pytree.py:630] <enum 'ScaleCalculationMode'> is an Enum subclass and is now natively supported by torch.compile as an opaque value type. Calling register_constant() on Enum subclasses is deprecated and will be an error in a future release.
> Unable to import `torchao` Tensor objects. This may affect loading checkpoints serialized with `torchao`

However the training is continuing, loss looking like it is working. If it fails I inform you here. It is much faster then before the latest update, so I am wondering...

ps. krea2 int8convrot lora/96rank ~3000 images.

**Comment by GreatPipisya** [comment]
is there any way to roll back to a working version?

**Comment by Bigfield77** [comment]
> is there any way to roll back to a working version?

what worked for me:  git reset --hard 0e1784176708e351ae664002a53baa585b5949fb

then pip install -r requirements.txt from your venv

**Comment by GreatPipisya** [comment]
> > is there any way to roll back to a working version?
> 
> what worked for me: git reset --hard [0e17841](https://github.com/ostris/ai-toolkit/commit/0e1784176708e351ae664002a53baa585b5949fb)
> 
> then pip install -r requirements.txt from your venv

Thanks!

**Comment by g3n3rativ3** [comment]
> [@g3n3rativ3](https://github.com/g3n3rativ3) what GPU are you running on, if i may ask?

Rtx pro 4500 blackwell (plus a mix of RAM modules totaling 160 GB at 3600 MT/s,)

**Comment by astragartist** [comment]
I guess this is the reason why there is no way to run it in runpod without issues. Hope it will be somehow fixed anytime soon. Thanks in advance!

**Comment by g3n3rativ3** [comment]
updated ! I think it is a rollback to torchao 0.10.0 if I remember what was written in the log.

**Comment by jaretburkett** [comment]
Fixed in 038f24e
